cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkProgramRunner.java

```diff
@@ -202,33 +202,27 @@ public ProgramClassLoader createProgramClassLoader(CConfiguration cConf, File di
   }
 
   /**
-   * Creates a {@link Closeable} for closing the ClassLoader of this {@link SparkProgramRunner}. The
+   * Closes the ClassLoader of this {@link SparkProgramRunner}. The
    * ClassLoader needs to be closed because there is one such ClassLoader created per program execution by
-   * the {@link SparkProgramRuntimeProvider} to support concurrent Spark program execution in the same JVM. The
-   * {@link Closeable#close()} method will be called when the program being executed through this program runner
-   * is completed.
+   * the {@link SparkProgramRuntimeProvider} to support concurrent Spark program execution in the same JVM.
    */
-  private Closeable createClassLoaderCloseable(final Id.Program programId, final RunId runId) {
+  @Override
+  public void close() throws IOException {
     final ClassLoader classLoader = getClass().getClassLoader();
-    return new Closeable() {
+    Thread t = new Thread("spark-program-runner-delay-close") {
       @Override
-      public void close() throws IOException {
-        Thread t = new Thread("delay-close-thread-" + programId + "-" + runId) {
-          @Override
-          public void run() {
-            // Delay the closing of the ClassLoader because Spark, which uses akka, has an async cleanup process
-            // for shutting down threads. During shutdown, there are new classes being loaded.
-            Uninterruptibles.sleepUninterruptibly(2, TimeUnit.SECONDS);
-            if (classLoader instanceof Closeable) {
-              Closeables.closeQuietly((Closeable) classLoader);
-              LOG.debug("Closed ClassLoader for SparkProgramRunner of {}, run {}.", programId, runId);
-            }
-          }
-        };
-        t.setDaemon(true);
-        t.start();
+      public void run() {
+        // Delay the closing of the ClassLoader because Spark, which uses akka, has an async cleanup process
+        // for shutting down threads. During shutdown, there are new classes being loaded.
+        Uninterruptibles.sleepUninterruptibly(2, TimeUnit.SECONDS);
```

Comment:
no need to make this configurable, right? Is 2 seconds sufficient always - guessing there is no reliable way of knowing when that async cleanup process is completed (listener or anything like that)?